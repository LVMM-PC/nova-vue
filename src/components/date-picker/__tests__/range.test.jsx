import NovaDatePicker from '@/components/date-picker/NovaDatePicker';
import { mount } from '@vue/test-utils';
import dayjs from 'dayjs';
import flushPromises from 'flush-promises';
import zhTW from '@/locales/lang/zh-TW';

describe('NovaDatePicker.vue type:range', () => {
  it('prop locale should works', async () => {
    const wrapper = mount({
      data() {
        let startDate = dayjs('2020-01-01')
          .startOf('day')
          .toDate();
        let endDate = dayjs('2020-12-31')
          .startOf('day')
          .toDate();
        return {
          someRange: [startDate, endDate]
        };
      },
      render() {
        return (
          <NovaDatePicker
            type="range"
            locale={zhTW}
            holiday={{}}
            value={this.someRange}
          />
        );
      }
    });
    wrapper.vm.$children[0].open('start');
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it('disabled date', async function() {
    function disabledDate(current) {
      let someDate = dayjs('2020-01-15')
        .startOf('day')
        .toDate();
      return current > someDate;
    }

    const wrapper = mount({
      data() {
        let startDate = dayjs('2020-01-01')
          .startOf('day')
          .toDate();
        let endDate = dayjs('2020-12-31')
          .startOf('day')
          .toDate();
        return {
          someRange: [startDate, endDate]
        };
      },
      render() {
        return (
          <NovaDatePicker
            type="range"
            holiday={{}}
            value={this.someRange}
            disabledDate={disabledDate}
          />
        );
      }
    });
    wrapper.vm.$children[0].open('start');
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });
});
