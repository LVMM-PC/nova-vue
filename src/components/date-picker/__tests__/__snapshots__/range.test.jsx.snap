// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NovaDatePicker.vue type:range disabled date 1`] = `
<div class="nova-date-picker is-open">
  <div class="nova-date-picker-toggle">
    <div class="nova-date-picker-inner"><input autocomplete="off" type="text" readonly="readonly" placeholder="请选择日期" class="nova-date-picker-input"> <span class="nova-date-picker-icon"></span></div>
  </div>
  <!---->
  <div data-mock="client-only">
    <div data-mock="mounting-portal" mountto="#portal-target" append="">
      <div class="nova-dropdown nova-date-picker-dropdown nova-slide-up-enter nova-slide-up-enter-active" style="left: 0px; top: 0px;">
        <div class="nova-date-picker-months">
          <div data-month-offset="0" data-range-index="0" class="nova-date-picker-month">
            <div class="nova-date-picker-header">
              <div title="上个月" class="nova-date-picker-prev">
                上个月
              </div>
              <div class="nova-date-picker-title">
                2020年 3月
              </div>
              <div title="下个月" class="nova-date-picker-next">
                下个月
              </div>
            </div>
            <div class="nova-date-picker-body">
              <div class="nova-date-picker-weeks">
                <div class="nova-date-picker-week nova-date-picker-sun">日</div>
                <div class="nova-date-picker-week nova-date-picker-mon">一</div>
                <div class="nova-date-picker-week nova-date-picker-tue">二</div>
                <div class="nova-date-picker-week nova-date-picker-wed">三</div>
                <div class="nova-date-picker-week nova-date-picker-thu">四</div>
                <div class="nova-date-picker-week nova-date-picker-fri">五</div>
                <div class="nova-date-picker-week nova-date-picker-sat">六</div>
              </div>
              <div class="nova-date-picker-dates">
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    1
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    2
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    3
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    4
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    5
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled is-today">
                  <div class="nova-date-picker-date-inner">
                    6
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    7
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    8
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    9
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    10
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    11
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    12
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    13
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    14
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    15
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    16
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    17
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    18
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    19
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    20
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    21
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    22
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    23
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    24
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    25
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    26
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    27
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    28
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    29
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    30
                  </div>
                </div>
                <div class="nova-date-picker-date is-disabled">
                  <div class="nova-date-picker-date-inner">
                    31
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    1
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    2
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    3
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    4
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    5
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    6
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    7
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    8
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    9
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    10
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-disabled">
                  <div class="nova-date-picker-date-inner">
                    11
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="nova-date-picker-tooltip" style="left: 0px; top: 0px; display: none;">

        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NovaDatePicker.vue type:range prop locale should works 1`] = `
<div class="nova-date-picker nova-date-picker-range">
  <!---->
  <div class="nova-date-picker-toggle">
    <div class="nova-date-picker-inner nova-date-picker-range-start"><input autocomplete="off" type="text" readonly="readonly" placeholder="開始日期" class="nova-date-picker-input"> <span class="nova-date-picker-icon"></span></div>
    <div class="nova-date-picker-inner nova-date-picker-range-end"><input autocomplete="off" type="text" readonly="readonly" placeholder="結束日期" class="nova-date-picker-input"> <span class="nova-date-picker-icon"></span></div>
  </div>
  <div data-mock="client-only">
    <div data-mock="mounting-portal" mountto="#portal-target" append="">
      <div class="nova-dropdown nova-date-picker-dropdown" style="left: 0px; top: 0px; display: none;">
        <div class="nova-date-picker-months">
          <div data-month-offset="0" data-range-index="0" class="nova-date-picker-month">
            <div class="nova-date-picker-header">
              <div title="上個月" class="nova-date-picker-prev">
                上個月
              </div>
              <div class="nova-date-picker-title">
                2020年 3月
              </div>
              <div title="下個月" class="nova-date-picker-next is-hidden">
                下個月
              </div>
            </div>
            <div class="nova-date-picker-body">
              <div class="nova-date-picker-weeks">
                <div class="nova-date-picker-week nova-date-picker-sun">日</div>
                <div class="nova-date-picker-week nova-date-picker-mon">一</div>
                <div class="nova-date-picker-week nova-date-picker-tue">二</div>
                <div class="nova-date-picker-week nova-date-picker-wed">三</div>
                <div class="nova-date-picker-week nova-date-picker-thu">四</div>
                <div class="nova-date-picker-week nova-date-picker-fri">五</div>
                <div class="nova-date-picker-week nova-date-picker-sat">六</div>
              </div>
              <div class="nova-date-picker-dates">
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    1
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    2
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    3
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    4
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    5
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range is-today">
                  <div class="nova-date-picker-date-inner">
                    6
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    7
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    8
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    9
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    10
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    11
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    12
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    13
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    14
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    15
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    16
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    17
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    18
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    19
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    20
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    21
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    22
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    23
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    24
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    25
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    26
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    27
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    28
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    29
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    30
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    31
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    1
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    2
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    3
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    4
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    5
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    6
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    7
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    8
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    9
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    10
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    11
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div data-month-offset="1" data-range-index="0" class="nova-date-picker-month">
            <div class="nova-date-picker-header">
              <div title="上個月" class="nova-date-picker-prev is-hidden">
                上個月
              </div>
              <div class="nova-date-picker-title">
                2020年 4月
              </div>
              <div title="下個月" class="nova-date-picker-next">
                下個月
              </div>
            </div>
            <div class="nova-date-picker-body">
              <div class="nova-date-picker-weeks">
                <div class="nova-date-picker-week nova-date-picker-sun">日</div>
                <div class="nova-date-picker-week nova-date-picker-mon">一</div>
                <div class="nova-date-picker-week nova-date-picker-tue">二</div>
                <div class="nova-date-picker-week nova-date-picker-wed">三</div>
                <div class="nova-date-picker-week nova-date-picker-thu">四</div>
                <div class="nova-date-picker-week nova-date-picker-fri">五</div>
                <div class="nova-date-picker-week nova-date-picker-sat">六</div>
              </div>
              <div class="nova-date-picker-dates">
                <div class="nova-date-picker-date is-prev is-in-range">
                  <div class="nova-date-picker-date-inner">
                    29
                  </div>
                </div>
                <div class="nova-date-picker-date is-prev is-in-range">
                  <div class="nova-date-picker-date-inner">
                    30
                  </div>
                </div>
                <div class="nova-date-picker-date is-prev is-in-range">
                  <div class="nova-date-picker-date-inner">
                    31
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    1
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    2
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    3
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    4
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    5
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    6
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    7
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    8
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    9
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    10
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    11
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    12
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    13
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    14
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    15
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    16
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    17
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    18
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    19
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    20
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    21
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    22
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    23
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    24
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    25
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    26
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    27
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    28
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    29
                  </div>
                </div>
                <div class="nova-date-picker-date is-in-range">
                  <div class="nova-date-picker-date-inner">
                    30
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    1
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    2
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    3
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    4
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    5
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    6
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    7
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    8
                  </div>
                </div>
                <div class="nova-date-picker-date is-next is-in-range">
                  <div class="nova-date-picker-date-inner">
                    9
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="nova-date-picker-tooltip" style="left: 0px; top: 0px; display: none;">

        </div>
      </div>
    </div>
  </div>
</div>
`;
